docs/application/dotnet/guides/applications/application_lifecycle.md

```diff
@@ -0,0 +1,194 @@
+                TextColor = Color.Blue,
+                PointSize = 12.0f,
+                HeightResizePolicy = ResizePolicyType.FillToParent,
+                WidthResizePolicy = ResizePolicyType.FillToParent
+            };
+
+            Window.Instance.GetDefaultLayer().Add(test);
+        }
+```
+
+`OnTerminate()` method logs information about the application termination.
+
+```csharp
+        protected override void OnTerminate()
+        {
+            Tizen.Log.Info("LifecycleTest", "OnTerminate()");
+            base.OnTerminate();
+        }
+```
+
+`OnPause()` method is called when application is suspended but not killed. Then it goes into the `Paused` state. To stop reading data from the `Accelerometer`, `SensorAccelerometer.Stop()` method is used. Now the application reduces resource usage. The main application window goes now into the background. UI events are not received but the data event handler have to be stopped manually.
+
+```csharp
+        protected override void OnPause()
+        {
+            Tizen.Log.Info("LifecycleTest", "OnPause()");
+            base.OnPause();
+
+            if (SensorAccelerometer != null) SensorAccelerometer.Stop();
+        }
+```
+
+The `Tizen` operating system may resume `Lifecycle` application if it not was terminated. Data updating is restored using the `SensorAccelerometer.Start()` method. The application main window now should be visible.
+
+```csharp
+        protected override void OnResume()
+        {
+            Tizen.Log.Info("LifecycleTest", "OnResume()");
+            base.OnResume();
+
+            if (SensorAccelerometer != null) SensorAccelerometer.Start();
+        }
+```
+
+The callback setup in function `InitSensors()` is called every one second. It prints debug information about a measured data from `Accelerometer` sensor. In the emulator case this values can be changed using `Control Panel` tool. For details please check [Log Viewer](/application/dotnet/guides/sensors/device-sensors.md)
+
+```csharp
+        void OnAccelerometer(object sender, AccelerometerDataUpdatedEventArgs args)
+        {
+            Tizen.Log.Info("LifecycleTest", string.Format("X: {0}, Y: {1}, Z: {2}", args.X, args.Y, args.Z));
+        }
+```
+
+To terminate the application `OnKeyEvent()` handler is implemented. When the back key is pressed `Exit()` method is called to terminate the application. To minimize the application the `Home` button should be pressed which is handled by the `Tizen` operating system. Then the `OnPause()` method will be called.
+
+```csharp
+        public void OnKeyEvent(object sender, Window.KeyEventArgs e)
+        {
+            if (e.Key.State == Key.StateType.Down && (e.Key.KeyPressedName == "XF86Back" || e.Key.KeyPressedName == "Escape"))
+            {
+                Exit();
+            }
+        }
+```
+
+The `Main` function code creates the `Program` instance and calls `Run` method to change application state to `Running` and call `OnCreate()` callback.
```

Comment:
In my opinion this would be  better: "The `Main` function code creates `Program` instance, calls Run method to change application state, and call `OnCreate()` callback?"

Reply:
@mihashco, Looks fine, just to confirm backticks (``) to be used for Run method.

Reply:
I think it is ok, In this case `Run` is a method defined in the app object.